### PR TITLE
image push

### PR DESCRIPTION
**Pull request by xpositivityx**

### CODE REVIEW

File: cmd/image.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+var imageCmd = &cobra.Command{
+	Use:   "image",
+	Short: "Container image integration Massdriver",
+	Long:  ``,
+}
+
+var imagePushCmd = &cobra.Command{
+	Use:   "push <namespace>/<image-name> <cloud-region> <massdriver-artifactId>",
```
Comment (coryodaniel) [inline cmd/image.go:25]: This has the positional arguments still

File: pkg/api2/conatiner_repository.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package api2
+
+import (
+	"context"
+
+	"github.com/Khan/genqlient/graphql"
+)
+
```
Comment (coryodaniel) [inline pkg/api2/conatiner_repository.go:8]: File spelling

**Review by coryodaniel — APPROVED** [review]
Two naming notes

File: cmd/image.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	RunE:  runImagePush,
+	Args:  cobra.ExactArgs(1),
+}
+
+var DockerBuildContext string
+var DockerfileName string
+var Tag string
+var ArtifactId string
+var ImageName string
+var Region string
```
Comment (chrisghill) [inline cmd/image.go:36]: Did you mean to make these public-scoped package variables? I don't see them read anywhere outside this file.

File: cmd/image.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+func runImagePush(cmd *cobra.Command, args []string) error {
+	setupLogging(cmd)
+
+	orgID := os.Getenv("MASSDRIVER_ORG_ID")
+	if orgID == "" {
+		log.Fatal().Msg("MASSDRIVER_ORG_ID must be set")
+	}
+
+	APIKey := os.Getenv("MASSDRIVER_API_KEY")
+	if orgID == "" {
```
Comment (chrisghill) [inline cmd/image.go:59]:
```suggestion
	if APIKey == "" {
```
Looks like this was a copy pasta error

File: pkg/image/docker_client.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+	return encodedAuth, nil
+}
+
+func setAuthUserNameByCloud(containerRepository *api2.ContainerRepository, auth *types.AuthConfig) error {
+	switch identifyCloudByRepositoryUri(containerRepository.RepositoryUri) {
+	case AWS:
+		auth.Username = "AWS"
+	case AZURE:
+		auth.Username = "00000000-0000-0000-0000-000000000000"
```
Comment (chrisghill) [inline pkg/image/docker_client.go:109]: WTF?? Is the the real life? Is this just fantasy?

**Review by chrisghill — APPROVED** [review]
A couple small suggestions.  Looks great!

File: pkg/api2/conatiner_repository.go (hunk 2 above)
Comment (xpositivityx) [reply pkg/api2/conatiner_repository.go:8]: I think this is the actual name: https://github.com/massdriver-cloud/massdriver-cli/blob/main/pkg/api2/deployment.go#L6

File: pkg/image/docker_client.go (hunk 5 above)
Comment (xpositivityx) [reply pkg/image/docker_client.go:109]: It is.

Comment (coryodaniel) [reply pkg/image/docker_client.go:109]: ALL BALLS

**Pull request merged**